Add everyone's first services to proxy

margotTenenbaum opened the pull request:
Hi @GrumpyCat0709 @Nick-Zabalza,

This PR covers the addition of everyone's first service to my proxy server, as well as new and improved testing to make sure that the services are loading, and that info from the db is making it into each service. Thanks!

Nick-Zabalza commented:
@margotTenenbaum 

Hi Elizabeth,

Page loads, I can see the components, and changing item number changes data consitently. Unfortunately, this time, the issues seem to outweigh what works. Those are:

1. While Kate's service loads, when you mouseover the main picture, you do not see the zoom view. I looked in the Element tab and that zoom view is being activated (DOM element that wasn't there before suddenly appearing upon mousing over large picture. This new DOM element also seems to have all the right styles and behavior associated with it). Since it does appear on my proxy, I can only assume the issue lies with point 2.

2. Your components do not line up in the middle of the page. While they don't expect pixel perfect CSS at this point, they do expect some CSS and the components not lining up in middle of page is quite eye-catching. i would put empty div tags on either side of a div that contains all of our components (but not the MODAL_ATTACH_POINT, that should still be contained by only the body). I would also put display: flex on that containing div along with any inner divs you add to further contain our components (Don't forget flex-direction, if needed). I think this is why Kate's service is not appearing properly on your proxy. I'm not 100% confident but it seems the way she coded her service, it expects to be wrapped in a container that has 'display' set to 'flex' and 'flex-direciton' set to the default 'row'. That is how i have it set up on my proxy and it works fine.

Question is: should Kate's service be able to display correctly without such a specific requirement? Or should she at least put in her ReadMe that her service's attach point needs the appropriate flex-direction and display styles? For now, I would suggest making a duplicate of your index.html file and placing the copy in a archive folder so that we can talk to Kate about next time we see her, before making the above (and below) changes to it.

3. Everything for my component displays correctly except for the modal. It doesn't appear at all. Upon further inspection, the problem appears to lie in the script tag for the index.html file. Lines 29-33 are currently inside the callback when they should be outside it (but still inside the script tag). I made the necassary change and my modal works as intended.

4. The SpecRunner test fails for the image from Kate's service, just like my test did. i know what the problem is but since this comment is already getting long, i will mention it when I reply to your comment in my proxy server. Other than this, your tests look great.

5. The ReadMe has several issues:
a. In Usage, point 2: You only mention using the desciption/etc. service but, at this point, that line should mention all three services your proxy currently funcitons with.
b. In Usage, point 4: you list it as itemId when it should be itemID
c.  Also in point 4, there is a typo for 'navigate'.

Looking forward to seeing the updates in action :)

margotTenenbaum commented:
Hi @Nick-Zabalza,

Thanks for your helpful code review! I believe I have addressed all of your concerns.
1. The zoom effect now works for the photoGallery service
2. Components are centered. (CSS improvements specific to my service will likely be ready for a separate PR tomorrow.)
3. Your modal now works.
4. I have modified the photoGallery test to not use a specific url for the time being.
5. README has been updated.

Nick-Zabalza commented:
@margotTenenbaum 

Looks great. You are free to merge.
